client/packages/programs/src/VaccineCourseEditModal/VaccineCourseEditModal.tsx

```diff
@@ -147,6 +145,7 @@ export const VaccineCourseEditModal: FC<VaccineCourseEditModalProps> = ({
       error(t('error.failed-to-save-vaccine-course'))();
       console.error(e);
     }
+    resetDraft();
```

Comment:
Do we want to reset if there's an issue saving? Sounds like potential data loss? Shouldn't this be next to the `success()` message?

Reply:
Yeah, fair point. I've moved it up into the "Success" condition: ac9f263f8e

(It actually previously had the `setIsDirty(false)` at the top of this method, so would have had a similar problem previously -- good spotting)